distributed/scheduler.py

```diff
@@ -2163,9 +2163,24 @@ def decide_worker_rootish_queuing_enabled(self) -> WorkerState | None:
             # All workers busy? Task gets/stays queued.
             return None
 
-        # Just pick the least busy worker.
-        # NOTE: this will lead to worst-case scheduling with regards to co-assignment.
-        ws = min(self.idle.values(), key=lambda ws: len(ws.processing) / ws.nthreads)
+        # Pick the least-busy worker. If multiple workers have no tasks, round-robin
+        # them to ensure different workers get tasks on a quiet cluster.
+        empties: list[WorkerState] = []
+        min_ws: WorkerState | None = None
+        min_v: float | None = None
+        for cws in dict.values(self.idle):
+            # ^ micro-optimization: `SortedDict` inherits from plain `dict`; iterating
+            # in non-sorted order is 10x faster and order doesn't matter here.
```

Comment:
The 10x number comes from here https://github.com/dask/distributed/pull/4925#discussion_r654679224

Reply:
Note that if we do https://github.com/dask/distributed/pull/7221 and/or https://github.com/dask/distributed/pull/6974, then `self.idle` doesn't need to be a `SortedDict` anymore, so we'd then get rid of this micro-optimization. The only place using the sortedness of `self.idle` is the current round-robin logic.

Reply:
xref https://github.com/dask/distributed/pull/7245